assets/queries/azureResourceManager/storage_account_allows_unsecure_transfer/query.rego

```diff
@@ -0,0 +1,47 @@
+package Cx
+
+import data.generic.common as common_lib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	[path, value] = walk(doc)
+
+	value.type == "Microsoft.Storage/storageAccounts"
+	resourceName := value.name
+
+	to_number(split(value.apiVersion, "-")[0]) < 2019
+
+	skValue := is_undefined(value)
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("resources.name={{%s}}%s", [resourceName, skValue]),
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": "resource with type 'Microsoft.Storage/storageAccounts' has the 'supportsHttpsTrafficOnly' property defined",
+		"keyActualValue": "resource with type 'Microsoft.Storage/storageAccounts' doesn't have 'supportsHttpsTrafficOnly' property defined",
+	}
+}
+
+CxPolicy[result] {
+	doc := input.document[i]
+	[path, value] = walk(doc)
+
+	value.type == "Microsoft.Storage/storageAccounts"
+	value.properties.supportsHttpsTrafficOnly == false
```

Comment:
I think here we could check if `value.properties.supportsHttpsTrafficOnly` is not true, since `supportsHttpsTrafficOnly` is not a required field. Also, maybe it should checks if it is undefined to define the search...
```suggestion
	not value.properties.supportsHttpsTrafficOnly
	skValue := is_undefined(value)
```

Reply:
I've made some adjustments to be more clear and added one more sample but the only way for it to trigger the query if API version >= 2019 is by defining `properties.supportsHttpsTrafficOnly` to false.. since the default is true